src/stratigraphy/util/find_description.py

```diff
@@ -24,16 +24,75 @@ def get_description_lines(lines: list[TextLine], material_description_rect: fitz
     Returns:
         list[TextLine]: The filtered lines.
     """
+    if not lines:
+        return []
     filtered_lines = [
         line
         for line in lines
         if line.rect.x0 < material_description_rect.x1 - 0.4 * material_description_rect.width
         if material_description_rect.contains(line.rect)
     ]
-
     return sorted([line for line in filtered_lines if line], key=lambda line: line.rect.y0)
 
 
+def get_description_blocks_from_layer_index(
+    layer_index_entries: list[TextLine], description_lines: list[TextLine]
+) -> list[TextBlock]:
+    """Divide the description lines into blocks based on the layer index entries.
+
+    Args:
+        layer_index_entries (list[TextLine]): The layer index entries.
+        description_lines (list[TextLine]): All lines constituting the material description.
+
+    Returns:
+        list[TextBlock]: The blocks of the material description.
+    """
+    blocks = []
+    line_index = 0
+    for layer_index_idx, _layer_index in enumerate(layer_index_entries):
+        # don't allow a layer above depth 0
+
+        next_layer_index = (
+            layer_index_entries[layer_index_idx + 1] if layer_index_idx + 1 < len(layer_index_entries) else None
+        )
+
+        matched_block = matching_blocks(description_lines, line_index, next_layer_index)
+        line_index += sum([len(block.lines) for block in matched_block])
+        blocks.extend(matched_block)
+
+    return blocks
+
+
+def matching_blocks(all_lines: list[TextLine], line_index: int, next_layer_index: TextLine | None) -> list[TextBlock]:
```

Comment:
Could become a method of `LayerIndexColumn`.

Reply:
If `next_layer_index` is not just an index but of type `TextLine | None` , then probably `next_layer` is a better name?

Reply:
I did not create a class object `LayerIndexEntry` similar to the entries of depthcolumns. The entries of the LayerIndex are simply TextLine objects.

What I mean here is the next layer index "entry". Therefore I called it next_layer_index. Does that make sense?

Moving forward and considering LayerIndexColumn as a special case of a DepthColum we might want to create the "LayerIndexEntry" object as well, and then this will be clearer. What do you think?